### PR TITLE
Calzetti fix

### LINKED ISSUE #142: Calzetti dust won't work
The implementation of the Calzetti dust law uses the N09 class from the external `dust_attenuation` module. However, this throws an exception if the wavelength is outside a particular range (which ours are). We need wrap some code around this that avoids the exception. For example, we could generate the attenuation curve only over the allowed portion (setting the rest to 1.0)?

### PR DESCRIPTION
**Pull request by aswinpvijayan**
## Issue Type
- Enhancement
Changed the Calzetti attenuation curve definition to remove infinities and nan at longer wavelengths.
I have also removed the from the requirements dust_attenuation package which is no longer used.

Closes #142 

## Checklist

- [x] I have added docstrings to all methods
- [x] I have added sufficient comments to all lines
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no pep8 errors
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/synthesizer/dust/attenuation.py [hunk 1]
```diff
@@ -39,26 +39,35 @@ def N09_tau(lam, slope, cent_lam, ampl, gamma):
...
-    k_lam = np.zeros_like(tmp_lam)
-
-    ok = (tmp_lam < 0.63)
-    k_lam[ok] = -2.156 + (1.509 / tmp_lam[ok]) \
-        - (0.198 / tmp_lam[ok] ** 2) \
-        + (0.011 / tmp_lam[ok] ** 3)
-    k_lam[~ok] = -1.857 + (1.040 / tmp_lam[~ok])
+    k_lam = np.zeros_like(lam_micron)
+
+    ok1 = (lam_micron >= 0.12) * (lam_micron < 0.63)
```
Comment (christopherlovell) [inline src/synthesizer/dust/attenuation.py:46]: could you provide some brief comments here as to what the `ok1`, `ok2` and `ok3` masks represent?

File: src/synthesizer/dust/attenuation.py [hunk 2]
```diff
@@ -39,26 +39,35 @@ def N09_tau(lam, slope, cent_lam, ampl, gamma):
...
+            + (0.011 / lam_micron[ok1]**3)
+        func = interpolate.interp1d(lam_micron[ok1], k_lam[ok1],
+                                fill_value="extrapolate")
+    if np.sum(ok2) > 0:
+        k_lam[ok2] = -1.857 + (1.040 / lam_micron[ok2])
+    if np.sum(ok3) > 0:
+        k_lam[ok3] = func(lam_micron[ok3])
+    
     k_lam = 4.05 + 2.659 * k_lam
+    k_v = 4.05 + 2.659 * (-2.156 + (1.509 / lam_v)
```
Comment (christopherlovell) [inline src/synthesizer/dust/attenuation.py:61]: Do these values come from Noll+09 or Calzetti+00 ?

**Review by christopherlovell — CHANGES_REQUESTED** [review]

**Comment by WillJRoper** [comment]
Does this close #142?

File: src/synthesizer/dust/attenuation.py (hunk 1 above)
Comment (aswinpvijayan) [reply src/synthesizer/dust/attenuation.py]: Added description

File: src/synthesizer/dust/attenuation.py (hunk 2 above)
Comment (aswinpvijayan) [reply src/synthesizer/dust/attenuation.py:65]: These are from Calzetti, the UV bump is the addition from Noll+09

**Review by christopherlovell — APPROVED** [review]

**Pull request merged**